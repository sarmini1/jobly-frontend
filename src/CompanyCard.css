.CompanyCard-card {
    padding: 3%;
    color: black;
    background: rgba( 255, 255, 255);
    box-shadow: 0 8px 32px 0 rgba(31, 38, 135, 0.096);
    backdrop-filter: blur( 16.0px );
    -webkit-backdrop-filter: blur( 16.0px );
    border-radius: 10px;
}

.CompanyCard {
    margin: 2%;
    font-family: 'Comfortaa', cursive;    
}

.CompanyCard:hover{
    box-shadow: 10px 5px 5px rgba(37, 37, 37, 0.507);
    color: rgb(65, 65, 153);
    border-radius: 5px;
}

b {
    margin-left: 20px;
}

.CompanyCard-logo {
    display: inline-block;
    max-height: 50px;
    margin-right: 50px;
}