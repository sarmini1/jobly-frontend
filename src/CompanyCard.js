import { Link } from "react-router-dom";
import { Card } from "react-bootstrap";
import './CompanyCard.css';


/**
 * CompanyCard Component
 * 
 * Props:
 * - company {}
 * 
 * CompanyList -> CompanyCard
 */
function CompanyCard({ company }) {
  const { name, handle, description, logoUrl } = company;

  return (
    <div className="CompanyCard">
      <Link to={`/companies/${handle}`} style={{ textDecoration: "none" }}>
        <Card className="CompanyCard-card">
          <Card.Title className="justify-content-between text-left">
            <div className="row">
              <b>{name}</b>
              {logoUrl && <img className="CompanyCard-logo ml-auto" src={logoUrl} alt={handle} />}
            </div>
          </Card.Title>
          <Card.Body className="text-left">
            <p>{description}</p>
          </Card.Body>
        </Card>
      </Link>
    </div>
  )
}

export default CompanyCard;