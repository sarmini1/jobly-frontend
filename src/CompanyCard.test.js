import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import CompanyCard from "./CompanyCard";

const testCompany = {
  "name": "Test Company Name",
  "handle": "test-company-handle",
  "description": "Great company to work for!",
  // "logoUrl":
};

it('mounts without crashing', function () {
  render(
    <MemoryRouter>
      <CompanyCard company={testCompany} />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { container } = render(
    <MemoryRouter>
      <CompanyCard company={testCompany} />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});

it("renders expected company info", function () {
  const { container } = render(
    <MemoryRouter>
      <CompanyCard company={testCompany} />
    </MemoryRouter>
  );
  expect(container).toContainHTML('<p>Great company to work for!</p>')
  expect(container).toContainHTML('<b>Test Company Name</b>')
});