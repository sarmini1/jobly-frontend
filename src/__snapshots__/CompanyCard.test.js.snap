// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="CompanyCard"
  >
    <a
      href="/companies/test-company-handle"
      style="text-decoration: none;"
    >
      <div
        class="CompanyCard-card card"
      >
        <div
          class="justify-content-between text-left card-title h5"
        >
          <div
            class="row"
          >
            <b>
              Test Company Name
            </b>
          </div>
        </div>
        <div
          class="text-left card-body"
        >
          <p>
            Great company to work for!
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
